XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Hands/BaseHandController.cs

```diff
@@ -0,0 +1,236 @@
+        /// Gets the hand controller's ray.
+        /// </summary>
+        protected HandRay HandRay { get; } = new HandRay();
+
+        /// <inheritdoc />
+        public bool IsTracked => TrackingState == TrackingState.Tracked;
+
+        /// <inheritdoc />
+        public bool IsInPointingPose => HandRay.ShouldShowRay;
+
+        /// <summary>
+        /// Gets the total joint count supported by this hand controller.
+        /// </summary>
+        public static readonly int JointCount = Enum.GetNames(typeof(TrackedHandJoint)).Length;
+
+        /// <summary>
+        /// Gets the current palm normal of the hand controller.
+        /// </summary>
+        protected Vector3 PalmNormal => TryGetJointPose(TrackedHandJoint.Palm, out MixedRealityPose pose) ? -pose.Up : Vector3.zero;
+
+        /// <inheritdoc />
+        public IReadOnlyDictionary<TrackedHandJoint, MixedRealityPose> JointPoses => jointPoses;
+
+        /// <inheritdoc />
+        public IReadOnlyDictionary<TrackedHandBounds, Bounds> Bounds => bounds;
+
+        /// <summary>
+        /// The Mixed Reality Controller default interactions.
+        /// </summary>
+        /// <remarks>A single interaction mapping works for both left and right controllers.</remarks>
+        public override MixedRealityInteractionMapping[] DefaultInteractions => new[]
+        {
+            new MixedRealityInteractionMapping(0, "Spatial Pointer", AxisType.SixDof, DeviceInputType.SpatialPointer, MixedRealityInputAction.None),
+            new MixedRealityInteractionMapping(1, "Spatial Grip", AxisType.SixDof, DeviceInputType.SpatialGrip, MixedRealityInputAction.None),
+            new MixedRealityInteractionMapping(2, "Select", AxisType.Digital, DeviceInputType.Select, MixedRealityInputAction.None),
+            new MixedRealityInteractionMapping(3, "Grab", AxisType.SingleAxis, DeviceInputType.TriggerPress, MixedRealityInputAction.None),
+            new MixedRealityInteractionMapping(4, "Index Finger Pose", AxisType.SixDof, DeviceInputType.IndexFinger, MixedRealityInputAction.None),
+        };
+
+        /// <inheritdoc />
+        public override MixedRealityInteractionMapping[] DefaultLeftHandedInteractions => DefaultInteractions;
+
+        /// <inheritdoc />
+        public override MixedRealityInteractionMapping[] DefaultRightHandedInteractions => DefaultInteractions;
+
+        /// <summary>
+        /// Controller constructor.
+        /// </summary>
+        /// <param name="trackingState">The controller's tracking state.</param>
+        /// <param name="controllerHandedness">The controller's handedness.</param>
+        /// <param name="inputSource">Optional input source of the controller.</param>
+        /// <param name="interactions">Optional controller interactions mappings.</param>
+        public BaseHandController(TrackingState trackingState, Handedness controllerHandedness, IMixedRealityInputSource inputSource = null, MixedRealityInteractionMapping[] interactions = null)
```

Comment:
Constructors should be at the top of the class.

Reply:
Where is this defined? Seems like a rather uncommon convention.

Reply:
It's the pattern we've been using throughout the codebase. I'll double check to see if it's in the coding guidelines.